odoo/tools/pdf.py

```diff
@@ -124,12 +126,23 @@ def addAttachment(self, name, data, subtype=""):
         :param name: The name of the attachement
         :param data: The data of the attachement
         :param subtype: The mime-type of the attachement. This is required by PDF/A, but not essential otherwise.
-        It should take the form of "/xxx%2Fxxx". E.g. for "text/xml": "/text%2Fxml"
+        It should take the form of "/xxx#2Fxxx". E.g. for "text/xml": "/text#2Fxml"
         """
+        # In case we receive a subtype that is not in the correct format, for example "application/xml",
+        # Change it to a format supported by the PDF files "/application#2Fxml"
+        adapted_subtype = subtype
+        if subtype and subtype[0] != '/':
+            adapted_subtype = '/' + subtype.replace('/', '#2F')
+
+        if not REGEX_SUBTYPE.match(adapted_subtype):
+            # The subtype still does not match the correct format, so we will not add it to the document
+            _logger.warning('Attempt to add an attachment with the incorrect subtype %s. The subtype will be ignored.', subtype)
```

Comment:
enclosing the subtype with delimiters helps with debugging, especially when the value can be an empty string or a space ;)

In this case you might need to avoid validating an empty subtype with the regex.


```suggestion
            _logger.warning("Attempt to add an attachment with incorrect subtype: '%s'. The subtype will be ignored.", subtype)
```

Reply:
I see, thank you

And indeed, we don't want to raise a warning each time we print something without any subtype.
I changed the code a little to use another regex to match on the mimetype format strings (instead of just checking the first character) and only do all that if we have a subtype given.